pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/mutable/provider/MutableForwardIndexProvider.java

```diff
@@ -0,0 +1,26 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.segment.spi.index.mutable.provider;
+
+import org.apache.pinot.segment.spi.index.mutable.MutableForwardIndex;
+
+
+public interface MutableForwardIndexProvider {
+  MutableForwardIndex newIndex(MutableIndexContext.Forward context);
```

Comment:
Consider renaming it to `newForwardIndex` to be consistent with the immutable provider interfaces? Same for other files

Reply:
I was thinking of changing the names of the other methods since the overloads can all be disambiguated by the parameter types, I felt this might be a neater convention. The other methods are not widely used yet so I don't think it's too late to change them.

Reply:
This is subjective, but IMO `newForwardIndex` reads better and less ambiguous. It's clear here in the interface definition because of the parameter type, but it will be a bit ambiguous in places where it's being used and the parameter name doesn't have word forward in it - like the one you have here `context`.

Reply:
OK